ratatui-widgets/src/chart.rs

```diff
@@ -395,6 +481,42 @@ impl<'a> Dataset<'a> {
         self
     }
 
+    /// Sets a list of value ranges that will modify the color of the line
+    ///
+    /// Used to specify a range of values that will be drawn in another color.
+    /// If the value does not fall inside any range specified the color will fallback to the style
+    /// foreground color.
+    ///
+    /// Has no effect in any [`GraphType`] other than [line](GraphType::Line)
+    ///
+    /// This is a fluent setter method which must be chained or used as it consumes self
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use ratatui::style::{Color, Style, Stylize};
+    /// use ratatui::widgets::{Dataset, GraphType, MultiColorLine};
+    ///
+    /// let dataset = Dataset::default()
+    ///     .graph_type(GraphType::Line)
+    ///     .multi_color_line(
+    ///         MultiColorLine::default()
+    ///             .add_color_range(0.0..1.0, Color::Green)
+    ///             .add_color_range(0.0..5.0, Color::Red), // equivalent to 1.0..5.0
+    ///     )
+    ///     .style(Style::default().cyan())
+    ///     .data(&[(0.0, 0.0), (1.0, 10.0)]);
+    /// ```
+    /// # Note
+    /// Ordering matters, ranges added first take precedence.
```

Comment:
style nit:

```suggestion
    /// ```
    ///
    /// # Note
    ///
    /// Ordering matters, ranges added first take precedence.
```